### PR TITLE
test: remove support for custom templates in NSTemplateSet

### PR DESCRIPTION
**Pull request by xcoulon**
See also:
- https://github.com/codeready-toolchain/api/pull/284
- https://github.com/codeready-toolchain/toolchain-common/pull/227
- https://github.com/codeready-toolchain/member-operator/pull/323
- https://github.com/codeready-toolchain/toolchain-e2e/pull/434
 
Fixes https://issues.redhat.com/browse/CRT-1382

Signed-off-by: Xavier Coulon <xcoulon@redhat.com>

### CODE REVIEW

File: go.mod [hunk 1]
```diff
@@ -31,4 +31,8 @@ require (
 	sigs.k8s.io/controller-runtime v0.8.3
 )
 
+replace github.com/codeready-toolchain/api => github.com/xcoulon/api v0.0.0-20220106110828-7a9658006b9d
+
+replace github.com/codeready-toolchain/toolchain-common => github.com/xcoulon/toolchain-common v0.0.0-20220106135935-9e2e34caa1ab
```
Comment (alexeykazakov) [inline go.mod:36]: Reminder.

**Review by alexeykazakov — APPROVED** [review]

**Review by rajivnathan — APPROVED** [review]
Looks good, just noticed a comment that I think should be removed too

**Review by MatousJobanek — APPROVED** [review]

File: go.mod (hunk 1 above)
Comment (xcoulon) [reply go.mod]: addressed in 4e73184

File: controllers/templateupdaterequest/templateupdaterequest_controller.go [hunk 2]
```diff
@@ -184,38 +184,20 @@ func (r Reconciler) updateTemplateRefs(logger logr.Logger, tur toolchainv1alpha1
...
-				if ns.Template != "" {
-					t := namespaceType(ns.TemplateRef)
-					logger.Info("retainining the custom namespace template", "type", t)
-					namespaces[t] = ns
-				}
-			}
 			// now, add the new templateRefs, unless there's a custom template in use
 			for _, ns := range tur.Spec.Namespaces {
 				t := namespaceType(ns.TemplateRef)
 				// don't override the custom template
```
[the remark this answers is not in the document]
Comment (xcoulon) [reply controllers/templateupdaterequest/templateupdaterequest_controller.go]: indeed! done in e76d4b0

**Pull request merged**